services/ui/ws/window_tree.cc

```diff
@@ -1974,6 +2015,19 @@ void WindowTree::SetCursor(uint32_t change_id,
     }
   }
 
+  // In external window mode, cursor changes are forwarded to
+  // the visible root for the client associated ws::Display.
+  if (window_server_->IsInExternalWindowMode()) {
+    WindowManagerDisplayRoot* display_root =
+        GetWindowManagerDisplayRoot(window);
+    if (display_root && display_root->GetClientVisibleRoot() == window) {
+      Display* display = GetDisplay(window);
+      DCHECK(display);
+
+      window = display_root->root();
```

Comment:
Is it valid to do so? Before this, the window is checked against the access policy and after that a root of that window is taken and the cursor is set. Maybe move it to line 1992? I guess it shouldn't fail as long as it's still the owner of the window who instantiates this change.